### PR TITLE
Combine the two controllers into a single pod

### PR DESCRIPTION
**Pull request by hectorj2f**
#### Summary

Fixes: https://github.com/sigstore/policy-controller/issues/631

#### Documentation

Unify our two main controllers into a single pod

### CODE REVIEW

**Comment by hectorj2f** [comment]
Locally works but it seems there is something going on on the github actions.

**Comment by anderssonw** [comment]
@hectorj2f Let me know if you need any particular help here. Would love to get this through before we do further work on our use of the controller, as this greatly reduces noices :)

**Comment by hectorj2f** [comment]
Thanks @anderssonw! I'm still investigating the issue but I have an idea of what is happening.

**Comment by hectorj2f** [comment]
This PR needs this bug fix to work https://github.com/knative/pkg/pull/2731.

File: config/501-policy-webhook-configurations.yaml [hunk 1]
```diff
@@ -36,16 +36,10 @@ webhooks:
...
     matchPolicy: Equivalent
     name: validating.clusterimagepolicy.sigstore.dev
     sideEffects: None
 ---
-apiVersion: v1
-kind: Secret
-metadata:
-  name: policy-webhook-certs
-  namespace: cosign-system
-# The data is populated at install time.
```
Comment (cpanato) [inline config/501-policy-webhook-configurations.yaml:51]: this is not needed anymore?

File: config/webhook.yaml [hunk 2]
```diff
@@ -11,7 +11,7 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
+---
```
Comment (cpanato) [inline config/webhook.yaml:14]: do we need this extra ticks?

**Review by cpanato — DISMISSED** [review]
lgtm

just a few nit comments

File: config/501-policy-webhook-configurations.yaml (hunk 1 above)
Comment (hectorj2f) [reply config/501-policy-webhook-configurations.yaml:51]: Yes, we are getting rid and unifying all the needs to one single pod.

**Comment by hectorj2f** [comment]
> but the title is in a single container instead of single pod right?

@cpanato  It is in a single pod. We had two pods before.

File: config/webhook.yaml (hunk 2 above)
Comment (hectorj2f) [reply config/webhook.yaml:14]: We need it to construct the yaml file with all the different yamls.

**Review by vaikas — APPROVED** [review]

**Pull request merged**